core/sodasql/cli/ingest.py

```diff
@@ -0,0 +1,189 @@
+"""
+CLI commands to ingest test results from various sources into the Soda cloud.
+"""
+
+from __future__ import annotations
+
+import datetime as dt
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Iterator
+
+from sodasql.__version__ import SODA_SQL_VERSION
+from sodasql.scan.parser import Parser
+from sodasql.scan.scan_builder import (
+    build_warehouse_yml_parser,
+    create_soda_server_client,
+)
+from sodasql.scan.test import Test
+from sodasql.scan.test_result import TestResult
+
+
+
+def create_dbt_run_result_to_test_result_mapping(
+    test_nodes: dict[str, CompiledSchemaTestNode],
+    run_results: list[RunResultOutput],
+):
+    """
+    Create a mapping with which test result belong to which run result.
+
+    Parameters
+    ----------
+    model_nodes : Dict[str: ParsedModelNode]
+        The parsed model nodes.
+    test_nodes : Dict[str: CompiledSchemaTestNode]
+        The compiled schema test nodes.
+    run_results : List[RunResultOutput]
+        The run results.
+
+    Returns
+    -------
+    out : Dict[str, set[ParseModelNode]]
+        A mapping from run result to test result.
+    """
+    from dbt.contracts.results import TestStatus
+    from dbt.contracts.graph.compiled import CompiledSchemaTestNode
+
+    dbt_tests_with_soda_test = {
+        test_node.unique_id:
+        Test(
+            id=Parser.create_test_id(
+                test_expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
```

Comment:
Because the compiled sql can change from result to result (especially for user defined macro tests), pushing the expression in here will lead to an entirely new test to be created on soda SQL because of how our model works.

We should find another way to generate the ID. As far as I can see, we could just as well leave it as None, or something else.

Reply:
ah, oke, that is good to know. If can confirm that `None` is fine, than that is ok for me. Another option is the raw sql:

```
"raw_sql": "{{ test_accepted_values(**_dbt_schema_test_kwargs) }}{{ config(alias='accepted_values_stg_soda__scan') }}",
```

Reply:
the current `create_test_id` returns something like:

```
        "id": '{"column":"result","test_name":"test.soda.accepted_values_stg_soda__scan__result__pass_fail.81f"}',
```

Is that to be expected?

Reply:
For the ID, I think the raw SQL may be ok but then again it may change as well. I would prefer to go by the test name and associated column or so.

I'm not sure I understand the latest comment.

Reply:
The last comment shows what the [`create_test_id` method](https://github.com/sodadata/soda-sql/blob/dc31150e3de38060683d27ed1e5bff5b6ec2802e/core/sodasql/scan/parser.py#L357) generates for the test data. It was a fyi, even though more fields are mentioned in [`create_test_id`](https://github.com/sodadata/soda-sql/blob/dc31150e3de38060683d27ed1e5bff5b6ec2802e/core/sodasql/scan/parser.py#L357) for the test data the `column` and `test_name` are the only two that are included.